src/components/WizardPattern/WizardPattern.js

```diff
@@ -0,0 +1,215 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  noop,
+  bindMethods,
+  propExists,
+  Modal,
+  Wizard,
+  Icon,
+  Button
+} from '../../index';
+import WizardPatternBody from './WizardPatternBody';
+import { stepShape } from './WizardPatternConstants';
+
+class WizardPattern extends React.Component {
+  constructor() {
+    super();
+    bindMethods(this, [
+      'getActiveStep',
+      'getPrevStep',
+      'getNextStep',
+      'onBackClick',
+      'onNextClick',
+      'goToStep',
+      'shouldPreventGoToStep'
+    ]);
+  }
+
+  onNextClick() {
+    const { steps, activeStepIndex } = this.props;
+    this.goToStep(Math.min(activeStepIndex + 1, steps.length - 1));
+  }
+
+  getStep(index = this.props.activeStepIndex) {
+    return this.props.steps[index];
+  }
+
+  getPrevStep(relativeToIndex = this.props.activeStepIndex) {
+    return relativeToIndex > 0 && this.props.steps[relativeToIndex - 1];
+  }
+
+  getNextStep(relativeToIndex = this.props.activeStepIndex) {
+    const { steps } = this.props;
+    return relativeToIndex < steps.length - 1 && steps[relativeToIndex + 1];
+  }
+
+  onBackClick() {
+    this.goToStep(Math.max(this.props.activeStepIndex - 1, 0));
+  }
+
+  goToStep(newStepIndex) {
+    const {
+      onStepChanged,
+      activeStepIndex,
+      onNext,
+      onBack,
+      steps
+    } = this.props;
+    const currentStep = steps[activeStepIndex];
+    if (this.shouldPreventGoToStep(newStepIndex)) return;
+    if (onStepChanged) onStepChanged(newStepIndex);
+    if (newStepIndex === activeStepIndex + 1) {
+      if (currentStep.onNext) currentStep.onNext();
+      if (onNext) onNext(newStepIndex);
+    }
+    if (newStepIndex === activeStepIndex - 1) {
+      if (onBack) onBack(newStepIndex);
+    }
+  }
+
+  shouldPreventGoToStep(newStepIndex) {
+    const { activeStepIndex, steps, nextStepDisabled } = this.props;
+    const targetStep = steps[newStepIndex];
+    const stepBeforeTarget = newStepIndex > 0 && steps[newStepIndex - 1];
+
+    const preventExitActive = this.getActiveStep().preventExit;
+    const preventEnterTarget = propExists(targetStep, 'preventEnter')
+      ? targetStep.preventEnter
+      : stepBeforeTarget && stepBeforeTarget.isInvalid;
+    const nextStepClicked = newStepIndex === activeStepIndex + 1;
+
+    return (
+      preventExitActive ||
+      preventEnterTarget ||
+      (nextStepClicked && nextStepDisabled)
+    );
+  }
+
+  render() {
+    const {
+      title,
+      show,
+      onHide,
+      onExited,
+      activeStepIndex,
+      steps,
+      nextStepDisabled,
+      stepButtonsDisabled,
+      cancelButtonText,
+      backButtonText,
+      nextButtonText,
+      closeButtonText
+    } = this.props;
+    const onFirstStep = activeStepIndex === 0;
+    const onFinalStep = activeStepIndex === steps.length - 1;
+    const activeStepStr = (activeStepIndex + 1).toString();
+
+    const prevStepUnreachable = onFirstStep || this.getPrevStep(); // || ??
```

Comment:
unclear comments, maybe suitable for removal?

Reply:
oh oops! I think I forgot to finish writing some logic here.....

Reply:
yep. very glad i left these comments. They were placeholders for missing logic that I've now added.